social/strategies/django_strategy.py

```diff
@@ -63,7 +62,14 @@ def request_path(self):
 
     def request_port(self):
         """Port in use for this request"""
-        return self.request.META['SERVER_PORT']
+        try:  # django >= 1.9
+            return self.request.get_port()
+        except AttributeError:  # django < 1.9
+            host_parts = self.request.get_host().split(':')
+            try:
+                return host_parts[1]
+            except IndexError:
+                return None
```

Comment:
`USE_X_FORWARDED_PORT` was introduced in django 1.9, and if we're on the `except AttributeError:` path it means we're using an older version so we can't use that setting. In this case, [`get_host()` will respect the `USE_X_FORWARDED_HOST` setting](https://github.com/django/django/blob/stable/1.8.x/django/http/request.py#L75-L86), and the reverse proxy can be configured to pass the port along with the `X-Forwarded-Host` header.

If you prefer we can fallback to using `self.request.META['SERVER_PORT']` from `except IndexError:`. The behaviour would be the same, as the port would just be stripped in the python-saml code later on.